packages/core/src/electron-browser/menu/electron-main-menu-factory.ts

```diff
@@ -72,15 +75,32 @@ export class ElectronMainMenuFactory {
                 }
 
                 items.push({
+                    id: menu.id,
                     label: menu.label,
                     icon: menu.icon,
+                    type: this.commandRegistry.getToggledHandler(commandId) ? "checkbox" : "normal",
+                    checked: this.commandRegistry.isToggled(commandId),
                     enabled: true, // https://github.com/theia-ide/theia/issues/446
                     visible: true,
-                    click: () => this.execute(menu.action.commandId),
+                    click: () => this.execute(commandId),
                     accelerator
                 });
+                if (this.commandRegistry.getToggledHandler(commandId)) {
+                    toggledCommands.push(commandId);
+                }
             }
         }
+        this.preferencesService.onPreferenceChanged(() => {
+            const applicationMenu = electron.remote.Menu.getApplicationMenu();
+            if (applicationMenu) {
+                for (const item of toggledCommands) {
+                    if (this.commandRegistry.isVisible(item)) {
```

Comment:
wait, maybe this is a misunderstanding. my point was to check, if the (sub-)menu is shown, for which we want to request the `toggle` state. 

asking the command registry for visibility is the wrong call here, it would ask the handler if the item should be shown at all.

Reply:
Removed previous check.
If I understood you correctly you are talking about checking if the menu is shown, but here the whole menu object needs to be updated despite visible items.